### ISSUE TITLE
Gnocchi's statsd rejects Swift's reports with "Invalid sampling for ms: `0`, should be none"

### ISSUE THREAD

**Issue by zaitcev** [gnocchixyz/gnocchi#279]
OpenStack Swift supports reporting to StatsD, but all of its timing is reported with sample rate, resulting in this:

2017-07-25 00:15:09.686 6875 ERROR gnocchi.statsd [-] Unable to treat metric s02.object-server.REPLICATE.timing:1.90305709839|ms|@0.1: Invalid sampling for ms: `0`, should be none

There's no official documentation for the protocol, and the topmost community document does not permit sample rate for "|ms" type. However, the original StatsD in JavaScript supports this kind of use.

I have put together a patch to absorb these reports, but it appears that it's not enough: no new metrics named "timing_count" are reported (using `gnocchi metric list`). Here it is, in case:

https://github.com/zaitcev/gnocchi/commit/f9ff9c6a26db9c1503e717036b5d68a3229eed75

**Comment by jd** [comment]
Fair enough, I'm not against supporting it anyway. I don't think nobody ever used that code so you're just testing what I wrote as a PoC a couple of years ago. :)

At first glance your patch looks correct, you probably want to make sure `flush` is called correctly and does something. Metric should be attached to the resource id you configured as a `statsd.resource_id` in the conf file.
